FogCreek/Glitch-Community
# [WIP] Interesting lupin: notification + privacy settings page

## Links
* Remix link: https://interesting-lupin.glitch.me/
* Product spec: https://docs.google.com/document/d/1yHd6Jt86UmdiDttkdPclvjnYi16B8-fTZnEgQgCW9vM/edit#heading=h.3im6lap0txtw

## GIF/Screenshots:
![Screen Shot 2019-09-16 at 5 50 17 PM](https://user-images.githubusercontent.com/938722/64995909-7cb83e00-d8aa-11e9-8297-04506549fdfd.png)

## Changes:
* create settings page
* create redux slice for managing settings

## How To Test:
**NOTE: none of this is hooked up to an API yet; the changes will not persist after reload**

* log into https://interesting-lupin.glitch.me/
* visit https://interesting-lupin.glitch.me/settings
* you should be able to toggle individual preferences
* turning a section off should turn off all child settings, and 'gray out' the section
* turnning a setting back on should re-select whatever child settings were selected
* you should be able to mute users besides yourself
* you should be able to mute notifications on your own projects
* you should be able to un-mute a previously muted user or project

## Things left to do before deploying:
- [ ] build a working API
- [ ] make (shared) Switch component with polished design
- [x] add handshake icon to shared components
- [x] fix bugs in shared components (scroll container border, PopoverMenu docs)
- [ ] figure out how this integrates with existing(?) settings pages
